Application/EdFi.Ods.Tests/EdFi.Ods.Api/Controllers/TokenControllerTests.cs

```diff
@@ -1,1626 +1,1652 @@
+            controller.ControllerContext = controllerContext;
+
+            return controller;
+        }
+
+        public class Using_client_credentials_grant
+        {
+            [TestFixture]
+            public class With_valid_key_and_secret : TestFixtureBase
+            {
+                private IClientAppRepo _clientAppRepo;
+                private IApiClientAuthenticator _apiClientAuthenticator;
+                private TokenController _controller;
+
+                private ApiClient _suppliedClient;
+                private Guid _suppliedAccessToken;
+
+                private TimeSpan _suppliedTTL;
+                private IActionResult _actionResult;
+
+                protected override void Arrange()
+                {
+                    _suppliedClient = new ApiClient
+                    {
+                        ApiClientId = 1
+
+                    };
+
+                    _suppliedAccessToken = Guid.NewGuid();
+                    _suppliedTTL = TimeSpan.FromMinutes(30);
+
+                    _clientAppRepo = Stub<IClientAppRepo>();
+
+                    //// Simulate a successful lookup of the client id/secret
+                    A.CallTo(() => _clientAppRepo.GetClientAsync(A<string>._)).Returns(_suppliedClient);
+
+                    _apiClientAuthenticator = A.Fake<IApiClientAuthenticator>();
+
+                    var accessToken = new ClientAccessToken(_suppliedTTL);
+                    accessToken.ApiClient = _suppliedClient;
+                    accessToken.Id = _suppliedAccessToken;
+
+                    A.CallTo(() => _clientAppRepo.AddClientAccessTokenAsync(A<int>._, A<string>._))
+                        .Returns(accessToken);
+
+                    A.CallTo(() => _apiClientAuthenticator.TryAuthenticateAsync(A<string>._, A<string>._)).Returns(new ApiClientAuthenticator.AuthenticationResult { IsAuthenticated = true, ApiClientIdentity = new ApiClientIdentity { Key = "clientId" } });
+                    var _tokenRequestProvider = Stub<ClientCredentialsTokenRequestProvider>();
+                    _tokenRequestProvider = new ClientCredentialsTokenRequestProvider(_clientAppRepo, _apiClientAuthenticator);
+                    _controller = CreateTokenController(_tokenRequestProvider);
+                }
+
+                protected override void Act()
+                {
+                    _actionResult = _controller.Post(
+                               new TokenRequest
+                               {
+                                   Client_id = "clientId",
+                                   Client_secret = "clientSecret",
+                                   Grant_type = "client_credentials"
+
+                               }).Result;
+                }
+
+                [Assert]
+                public void Should_return_HTTP_status_of_OK()
+                {
+                    var result = (OkObjectResult)_actionResult;
```

Comment:
the asserts should be _actionResult.ShouldBeOfType<OkObjectResult>() (i cant remember the syntax please look it up).
_actionResult.StatusCode.ShouldBe(HttpStatusCode.Ok)